src/App.tsx

```diff
@@ -17,18 +59,39 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                query={query}
+                setQuery={setQuery}
+                filterByOption={filterOption}
+                setFilterByOption={setFilterOption}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {isLoading
+                ? <Loader />
+                : (
+                  <TodoList
+                    todos={handleFilter(todos, query, filterOption)}
+                    setShowModal={setShowModal}
+                    setSelectedTodo={setSelectedTodo}
+                    setUserId={setUserId}
+                    selectedTodo={selectedTodo}
+                  />
+                )}
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {showModal && (
+        <TodoModal
+          userId={userId as number}
```

Comment:
but it can be null, can't it?
try to figure out the solution without `as number`